src/WebJobs.Extensions.DurableTask/DurabilityProvider.cs

```diff
@@ -374,5 +374,17 @@ public Task PurgeOrchestrationHistoryAsync(DateTime thresholdDateTimeUtc, Orches
         {
             return this.GetOrchestrationServiceClient().PurgeOrchestrationHistoryAsync(thresholdDateTimeUtc, timeRangeFilterType);
         }
+
+        /// <summary>
+        /// Uses durability provider specific logic to verify whether a timespan for a timer
+        /// or a retry interval is allowed by the provider.
+        /// </summary>
+        /// <param name="timespan">The timespan for a timer or retry interval.</param>
+        /// <returns>A boolean indicating whether the time interval is valid and an error message if it is invalid.</returns>
+        public virtual (bool, string) CheckTimeInterval(TimeSpan timespan)
```

Comment:
I also suggest we consider a name like `ValidateTimerExpiration` to make it more clear what scenarios the timespan applies to.

Reply:
I didn't want to necessarily use TImer in the name, as it also applies to timeouts and retry intervals. Maybe `ValidateDelayTime`, but if you don't like that, I think that `ValidateTimerExpiration` still works.

Reply:
I mentioned Timers since our time constraints are only needed for scenarios where timers are used. But I guess one could argue that this may be too specific to the Azure storage provider. In any case, I like the new name you chose. :)